hw-pawlenko/src/hw11032021_executor_service_example/Main.java

```diff
@@ -0,0 +1,34 @@
+package hw11032021_executor_service_example;
+
+import java.io.*;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+
+public class Main {
+    static String file = "src/image-url.txt";
+    static String directory = "outImages/";
+
+    public static void main(String[] args) {
+        List<String> listUrl = new ArrayList<>();
+
+        try {
+            listUrl = Files.lines(Paths.get(file).toAbsolutePath()).collect(Collectors.toList());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        ExecutorService executor = Executors.newFixedThreadPool(10);
+        for (String url : listUrl) {
+            executor.submit(new ImageDownload(url, directory));
```

Comment:
It may be the case that your main thread is finished before all the tasks. You need to wait in the main thread whether all tasks are completed by checking the Future for each task, which is returned by the submit().